botcity/core/application/functions.py

```diff
@@ -21,30 +24,42 @@ def connect(backend=Backend.WIN_32, timeout=60000, **connection_selectors) -> Ap
             ](https://documentation.botcity.dev/frameworks/desktop/windows-apps/).
 
     Returns
-        app (Application): The Application instance.
+        app (Application | WindowSpecification): The Application/Window instance.
     """
     connect_exception = None
     start_time = time.time()
     while True:
         elapsed_time = (time.time() - start_time) * 1000
         if elapsed_time > timeout:
-            if connect_exception:
-                raise connect_exception
-            return None
+            break
         try:
             app = Application(backend=backend).connect(**connection_selectors)
             return app
         except Exception as e:
             connect_exception = e
             time.sleep(config.DEFAULT_SLEEP_AFTER_ACTION/1000.0)
 
+    if "path" in connection_selectors.keys():
+        connection_selectors.pop("path")
 
-def find_window(app: Application, waiting_time=10000, **selectors) -> WindowSpecification:
+    if not connection_selectors:
+        if connect_exception:
+            raise connect_exception
+        return None
+
+    app = Desktop(backend=backend).window(**connection_selectors)
+    if not app.exists():
+        raise WindowNotFoundError(f"Unable to find an app using these criteria: {connection_selectors}")
+    return app
```

Comment:
This implementation idea looks great, but I think we can possibly improve it.
As of now, it will try up to `timeout` milliseconds to connect to the Application and only if that fails it will try the fallback method with the `Desktop` approach.

While the `Desktop` approach was introduced to handle a corner case, I believe it could be added into the timeout loop in some way for us to also check using this method if the `Application` connection fails as an alternative to speed up and not required that applications relying on the `Desktop.window` method wait the full timeout period before they can connect.

What do you think?